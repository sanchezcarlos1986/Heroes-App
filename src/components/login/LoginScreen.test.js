import React from 'react';
import {LoginScreen} from '~components';
import '@testing-library/jest-dom';
import {mount} from 'enzyme';
import {AuthContext} from '~auth';
import {types} from '~types';

describe('LoginScreen', () => {
  let component;
  const history = {
    replace: jest.fn(),
  };

  const context = {
    dispatch: jest.fn(),
    user: {logged: false},
  };

  afterEach(() => jest.clearAllMocks());

  beforeEach(() => {
    component = mount(
        <AuthContext.Provider value={context}>
          <LoginScreen history={history} />
        </AuthContext.Provider>,
    );
  });

  it('should renders ok', () => {
    expect(component).not.toBeNull();
    expect(component).toMatchSnapshot();
  });

  it('should calls dispatch and lastPath should be /', () => {
    const expectedAction = {
      payload: {name: 'Carlos Sánchez'},
      type: types.login,
    };

    component.find('button').simulate('click');

    expect(context.dispatch).toBeCalledWith(expectedAction);
    expect(history.replace).toBeCalledWith('/');
  });

  it('should calls dispatch and lastPath should be /search', () => {
    const expectedAction = {
      payload: {name: 'Carlos Sánchez'},
      type: types.login,
    };

    const lastPath = '/search';

    localStorage.setItem('lastPath', lastPath);
    component.find('button').simulate('click');

    expect(context.dispatch).toBeCalledWith(expectedAction);
    expect(history.replace).toBeCalledWith(lastPath);
  });
});
