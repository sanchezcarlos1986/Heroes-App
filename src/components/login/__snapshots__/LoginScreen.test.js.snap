// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should renders ok 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
